.github/workflows/release.yml

```diff
@@ -0,0 +1,123 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'The version being released'
+        required: true
+        type: string
+      update-changelog:
+        description: 'Update CHANGELOG'
+        required: true
+        type: boolean
+        default: true
+      notification-channel:
+        description: 'Slack channel to use for notifications'
+        required: false
+        type: string
+        default: 'CUYKT2A73'
+
+env:
+  GO_TAGS: "release"
+
+jobs:
+  get-go-version:
+    runs-on: ubuntu-latest
+    outputs:
+      go-version: ${{ steps.get-go-version.outputs.go-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Determine Go version
+        id: get-go-version
+        # We use .go-version as our source of truth for current Go
+        # version, because "goenv" can react to it automatically.
+        run: |
+          echo "Building with Go $(cat .go-version)"
+          echo "::set-output name=go-version::$(cat .go-version)"
+
+  prepare-release:
+    needs: get-go-version
+    runs-on: ubuntu-latest
+    outputs:
+      build-ref: ${{ steps.commit-change-push.outputs.build-ref }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ needs.get-go-version.outputs.go-version }}
+
+      - name: Setup node and yarn
+        uses: actions/setup-node@v2
+        with:
+          node-version: "14"
+          cache-dependency-path: "ui/yarn.lock"
+
+      - name: Install Yarn
+        run: |
+          npm install -g yarn
+
+      - name: Install dependencies
+        run: |
+          make deps
+
+      - name: Update notification channel
+        if: ${{ github.event.inputs.notification-channel != '' }}
+        run: |
+          sed -i.bak -e 's|\(notification_channel * = *"\)[^"]*|\1${{ github.event.inputs.notification-channel }}|g' .release/ci.hcl
```

Comment:
Just noting that we hope to allow [richer syntax](https://hashicorp.atlassian.net/browse/RELTOOL-124?atlOrigin=eyJpIjoiNGU5NGM1YzlkOWEzNDc1NWE2YmY1ODI2ZDNjMTAzMjYiLCJwIjoiaiJ9) in the future - to make these sort of things easier to configure in the file.